src/coreclr/interpreter/compiler.cpp

```diff
@@ -1903,11 +2177,36 @@ int InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
     m_pEntryBB->stackHeight = 0;
     m_pCBB = m_pEntryBB;
 
+    //
+    // We will create two separated linked list of basic blocks, the blocks of funclets
+    // and the blocks of others.
+    //
+    // The reason of doing that is that the runtime requires the funclets to be separated from
+    // the body so that funclets are never inside try
```

Comment:
Funclets can be inside a `try`. For example:
```
try // TryA
{
    try // TryB
    {
        ...
    }
    finally
    {
        // This is a funclet that is inside (or more precisely - covered by) TryA
    }
}
catch
{
    ...
}
```

Reply:
The regular JIT duplicates the EH clauses as necessary to move funclets out-of-file while keeping them covered by try clauses. My example has two EH clauses in IL, but it has 4 EH clauses for the generated code due to this duplication.

Is the interpreter going to do the same?

Reply:
> The regular JIT duplicates the EH clauses as necessary to move funclets out-of-file while keeping them covered by try clauses.

It does not anymore. @filipnavara has removed creating them recently as the new EH didn't use them.

Reply:
When exception escapes a funclet, we have the "collided handling" mechanism that restores the stack frame iterator from the previous ExInfo and continues looking for handler where the previous EH left it. This is the same way as we have always had in nativeAOT.

Reply:
Ah ok, I have not realized that we got rid of the duplication completely. I thought that there is still some left.